### ISSUE TITLE
Allow passing vault pass as a value

### ISSUE THREAD

**Issue by aeons** [MeilleursAgents/terraform-provider-ansiblevault#47]
We don't have our vault pass in a file already, so to use this provider, we need to write our pass out to a file, to then read it back in again.

It would be beneficial if it was possible to supply the vault pass as a value, instead of a file.

**Comment by ViBiOh** [comment]
Hi @aeons,

**Comment by ViBiOh** [comment]
HI @aeons ,

Thank you for your interest in our provider.

We have recently add the feature of string as vault. Checkout our example [here](https://github.com/MeilleursAgents/terraform-provider-ansiblevault/blob/master/examples/terraform/data.tf#L11).

As @bdronneau said in #39 , we only manage "key/value" in the yaml form, so your unencrypted string should look like 

```yaml
secret_token: '1234abcd'
```

**Comment by aeons** [comment]
Hi @ViBiOh,

We already have our secrets in yaml format (for another tool), what I would like is to give the vault *password* as a value instead of as a path to a file that contains the password.

**Comment by ViBiOh** [comment]
Yes, it's possible with the `ansiblevault_string` data https://github.com/MeilleursAgents/terraform-provider-ansiblevault/blob/master/examples/terraform/data.tf#L11.

You pass the encrypted value (not a file) and can query value for decrypted string

**Comment by aeons** [comment]
Yeah, I get that.

I have my encrypted values in a file. That part is fine.

I do not have the password for the vault in a file. To use the provider, I need to pass the password in a file, and I would like to avoid that, and pass it as a value instead.

**Comment by ViBiOh** [comment]
Ah, I get it!

This approach is not recommended. Terraform files should be add to your versionning system (git) and if you add your ansible vault password in clear text, you'll have a security issue.

**Comment by trygvis** [comment]
This is where Ansible's executable secret files are useful, my passwords are encrypted on disk and decryptet and kept in memory only for the duration of the Ansible run.

**Comment by aeons** [comment]
Yeah, I know. We do not have the password in clear text. We have it in another tool that gets it in a secure way.
What I would like to do is call terraform like
```
terraform apply -var vault_pass=somesecretkey
```
And use the variable in terraform to give to the provider.

**Comment by trygvis** [comment]
Yeah, except then the password ends up in my bash history. And I already have a seamless solution with Ansible to decrypt the passwords.

If I want to shoe horn this in, I guess I can use the "external" Terraform provider to give me the password which in turn is used to decode the vault.

**Comment by aeons** [comment]
My response was in regard to @ViBiOh :)

Our setup is running this in CI, where it doesn't really leak. The executable pass file would work as well, but that is not implemented in the provider.

**Comment by bdronneau** [comment]
Hi @aeons,

PR will be open today for this feature.

**Comment by aeons** [comment]
Awesome, thanks!

**Comment by bdronneau** [comment]
PR is done, release will done soon

**Comment by aeons** [comment]
Closed by #53
